app/models/collection.rb

```diff
@@ -14,4 +14,18 @@ class Collection < ApplicationRecord
   def review_enabled?
     reviewers.present?
   end
+
+  state_machine initial: :first_draft do
+    event :begin_deposit do
+      transition first_draft: :depositing, version_draft: :depositing
+    end
+
+    event :deposit_complete do
+      transition depositing: :deposited
+    end
+
+    event :update do
+      transition [:deposited, :version_draft] => :version_draft
```

Comment:
What does this syntax mean? How is it different from the `begin_deposit` transition syntax?

Reply:
because this means you don't have to check `work.deposited?` you can always have this event, because it's a no-op to go from version_draft back to version_draft.

Reply:
I'm not sure I follow why we need this for collections (and not for works). Maybe a state diagram for collections would help.

Reply:
We could have it for works, but you moved the transition to the EventService and wrapped it with a conditional: https://github.com/sul-dlss/happy-heron/blob/a138efe8fa2ee4975914303cfa662c95e42d89a8/app/controllers/works_controller.rb#L49

Reply:
The conditional was there before the EventService came along: https://github.com/sul-dlss/happy-heron/commit/64b54490ca250aeb3b3d6fb34e4ac0f3057fc853#diff-26bc953ee940475a6382d93330c2315a15a75269bbf9ad0798c926f8c2b5a7c7L49